### PR TITLE
[bugfix] defining undefined variables in lizmapWMSRequest

### LINKED ISSUE #1924: Undefined variables in lizmapWMSRequest
### What is the bug?

There is 3 undefined variables in `lizmap/modules/lizmap/classes/lizmapWMSRequest.class.php` on master branch.

line 188 : `$data`
line 470 : `$gfiparams`
line 808 :  `$xmldata`

### PR DESCRIPTION
**Pull request by alagroy-42**
* Ticket : fixes #1924

### CODE REVIEW

**Review by laurentj — COMMENTED** [review]
It seems ok to me, at the syntax level. But @rldhont knows more than me if new values are correct

File: lizmap/modules/lizmap/classes/lizmapWMSRequest.class.php [hunk 1]
```diff
@@ -185,7 +185,7 @@ protected function getcontext()
             array('repository' => $this->repository->getKey(), 'project' => $this->project->getKey())
         );
         $sUrl = str_replace('&', '&amp;', $sUrl).'&amp;';
-        $data = preg_replace('/xlink\:href=".*"/', 'xlink:href="'.$sUrl.'&amp;"', $data);
+        $response->data = preg_replace('/xlink\:href=".*"/', 'xlink:href="'.$sUrl.'&amp;"', $response->data);
```
Comment (rldhont) [inline lizmap/modules/lizmap/classes/lizmapWMSRequest.class.php:188]:
```suggestion
        $data = $response->data;
        $data = preg_replace('/xlink\:href=".*"/', 'xlink:href="'.$sUrl.'&amp;"', $data);
```
I prefer to work as immutable even if it is not.
And change the line 93

**Review by rldhont — CHANGES_REQUESTED** [review]
I think you have to change line 188 and 193

**Comment by rldhont** [comment]
@alagroy-42 is this fix for release 3.4 or for 3.5 ?

**Comment by alagroy-42** [comment]
@rldhont For release 3.4, these variables exist and are not defined in the current version of the release.

**Comment by Gustry** [comment]
@alagroy-42 We usually do the bugfix on the lowest branch impacted. So you PR must target 3.4 and not master.
The commit will come back later on master.

**Pull request closed without merging**